crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -288,8 +288,6 @@ use std::{cell::UnsafeCell, marker::PhantomData};
 ///     _bar: With<Bar>,
 ///     _or: Or<(With<Baz>, Changed<Foo>, Added<Bar>)>,
 ///     _generic_tuple: (With<T>, Without<P>),
-///     #[world_query(ignore)]
-///     _tp: std::marker::PhantomData<(T, P)>,
```

Comment:
Do we have any other occurrence of `#[world_query(ignore)]` anywhere in the test or examples? So we are still sure that it always works and compiles.

Reply:
I think we don't, im not actually sure if we even need `world_query(ignore)` tbh :sweat_smile: its not much but everything worked before i removed all the `world_query(ignore)` fields

Reply:
I can imagine having some conditionally compiled members with generic arguments, for which one may need an ignored `PhantomData<T>` (for the case when such a member isn't compiled but a generic argument needs to be used).

Reply:
Conditional compilation is definitely something that I imagine being used in the apps that support a headless mode: for instance, multiplayer servers, where you want to skip client-only components. I agree that having them both conditional and generic is quite a rare scenario, but I'd say it would still be great to have the flexibility :)